src/main/kotlin/com/autonomousapps/internal/asm.kt

```diff
@@ -251,7 +250,8 @@ internal class ClassAnalyzer(private val logger: Logger) : ClassVisitor(ASM_VERS
   override fun visitAnnotation(descriptor: String?, visible: Boolean): AnnotationVisitor {
     log("ClassAnalyzer#visitAnnotation: descriptor=$descriptor visible=$visible")
     addClass(descriptor, ClassRef.Kind.ANNOTATION)
-    return annotationAnalyzer
+    return AnnotationAnalyzer(visible, logger, classes)
```

Comment:
Third bug was treating class references in runtime-retained annotations like they weren't needed at compile and runtime (they are).